ts-node-plugin/index.js

```diff
@@ -0,0 +1,38 @@
+const {transform} = require("../dist");
+
+const ModuleKindCommonJS = 1;
+
+/**
+ * ts-node transpiler plugin
+ */
+function create(createOptions) {
+  const {nodeModuleEmitKind} = createOptions;
+  const {module} = createOptions.service.config.options;
+  return {
+    transpile(input, transpileOptions) {
+      const {fileName} = transpileOptions;
+      const transforms = ["typescript"];
+      if (module === ModuleKindCommonJS || nodeModuleEmitKind === "nodecjs") {
+        transforms.push("imports");
+      }
+      if (fileName.endsWith(".tsx") || fileName.endsWith(".jsx")) {
+        transforms.push("jsx");
+      }
+
+      const {code, sourceMap} = transform(input, {
```

Comment:
Another thing to think about here is how to handle the TS `esModuleInterop` option, which is enabled by default in Sucrase but can be disabled via the Sucrase `enableLegacyTypeScriptModuleInterop` flag. I've been thinking about dropping support in the next breaking change, though now I'm a little worried about that causing issues for older TS projects that are using `ts-node` and want to try out the `sucrase: true` option. For now, I'll map the option correctly (with a test), and I'm leaning toward keeping the option indefinitely.